### PR TITLE
hw 8-9

### PR DESCRIPTION
**Pull request by CherednichenkoNikita**

### CODE REVIEW

**Review by ochedenis — APPROVED** [review]

File: src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return scanner.next();
+    }
+
+    public static List<Worker> comparatorForWorker(List<Worker> workerList) {
+        workerList.sort(new Comparator<Worker>() {
+            @Override
+            public int compare(Worker o1, Worker o2) {
+                return o1.getFullName().compareTo(o2.getFullName());
+            }
+        });
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java:49]: It's better to shift anonymous class to static nested class

File: src/main/java/com/java_essential/Cherednichenko/homework/lesson_9/Device.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Device)) return false;
+        Device device = (Device) o;
+        return Float.compare(device.price, price) == 0 &&
+                Objects.equals(manufacturer, device.manufacturer) &&
+                Objects.equals(serialNumber, device.serialNumber);
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_essential/Cherednichenko/homework/lesson_9/Device.java:44]: Avoid bad readable code, simplify to variant bellow
```suggestion
   @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        
        final Device other = (Device) obj;
        
        if (!Objects.equals(this.manufacturer, other.manufacturer)) {
            return false;
        }
        
        if (!Objects.equals(this.price, other.price)) {
            return false;
        }
        if (!Objects.equals(this.serialNumber, other.serialNumber)) {
            return false;
        }
        
        return true;
    }
```

File: src/main/java/com/java_essential/Cherednichenko/homework/lesson_9/Device.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(manufacturer, price, serialNumber);
+    }
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_essential/Cherednichenko/homework/lesson_9/Device.java:50]: You have violated the contract between equals and hashCode.
Change it to variant bellow.
```suggestion
   @Override
    public int hashCode() {
        int hash = 7;
        hash = 73 * hash + Objects.hashCode(manufacturer);
        hash = 73 * hash + Objects.hashCode(price);
        hash = 73 * hash + Objects.hashCode(serialNumber);
        return hash;
    }
```

File: src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package com.java_essential.Cherednichenko.homework.lesson_8.task_1;
+
+import java.util.*;
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java:3]: replace this import with single class import

File: src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                log.info(e);
+                System.out.println("Wrong year: " + name);
+            }
+        }
+        workerList = comparatorForWorker(workerList);
+        System.out.println("Result: ");
+        for (Worker worker : workerList) {
+            System.out.println(worker.getFullName());
+        }
+    }
```
Comment (yuriy-tsarenko) [inline src/main/java/com/java_essential/Cherednichenko/homework/lesson_8/task_1/Main.java:35]: Very big main method body! 
main method it's a access point of your app.

**Review by yuriy-tsarenko — CHANGES_REQUESTED** [review]
Fix bugs according to my comments in similar cases

**Pull request merged**